### PR TITLE
Adding design doc template

### PR DESCRIPTION
**Pull request by andrebriggs**

### CODE REVIEW

File: docs/commands/data.json [hunk 1]
```diff
@@ -191,10 +191,6 @@
         "description": "Name of the resource group to create new storage account when it does not exist",
         "required": true
       },
-      {
```
Comment (NathanielRose) [inline docs/commands/data.json]: Was this intentionally made to be modified?

**Review by NathanielRose — APPROVED** [review]

File: docs/commands/data.json (hunk 1 above)
Comment (andrebriggs) [reply docs/commands/data.json]: @NathanielRose we have a docs regeneration pre-commit hook that is updating the docs

Comment (dennisseah) [reply docs/commands/data.json]: @NathanielRose 
this is generated by pre-commits hooks. it is ok

**Comment by dennisseah** [comment]
I think we can add Appendix section where we list down (if any) url to documents/articles/guides that aid in writing this doc. what do you think?

**Comment by andrebriggs** [comment]
> I think we can add Appendix section where we list down (if any) url to documents/articles/guides that aid in writing this doc. what do you think?

@dennisseah Good idea. I'll update

**Review by samiyaakhtar — APPROVED** [review]

**Review by dennisseah — APPROVED** [review]

**Pull request merged**